### ISSUE TITLE
Reduce list-of-links issue (from CDC) to a clear test case

### ISSUE THREAD

**Issue by Mr0grog** [edgi-govdata-archiving/web-monitoring-processing#124]
The LXML-based visual diff (i.e. `html_diff_render`) of this change:

https://api.monitoring.envirodatagov.org/api/v0/pages/376d7f27-fa20-448a-a5ab-0ec0e78aa1ff/changes/f96bf124-57fd-46e8-a2e4-7aeef185d4ea..db9e5f92-8d16-4038-9b88-27ce723396ed

(View in the UI at: https://monitoring.envirodatagov.org/page/376d7f27-fa20-448a-a5ab-0ec0e78aa1ff/f96bf124-57fd-46e8-a2e4-7aeef185d4ea..db9e5f92-8d16-4038-9b88-27ce723396ed)

Does a pretty poor job with the list of alphabetical links. Here’s an example snippet:

```html
<div><ul><li><a href=\"https://www.cdc.gov/az/a.html\">A <ins> Link: https://www.cdc.gov/az/a.html </ins><li><ins><a href=\"https://www.cdc.gov/az/b.html\">B Link: https://www.cdc.gov/az/b.html </a></ins></li><ins> </ins><li><ins><a href=\"https://www.cdc.gov/az/c.html\">C Link: https://www.cdc.gov/az/c.html </a></ins></li>

...

<li><ins><a href=\"https://www.cdc.gov/az/0.html\"># Link: https://www.cdc.gov/az/0.html </a></ins></li> <del> Link: //www.cdc.gov/az/a.html </del><li><del><a href=\"//www.cdc.gov/az/b.html\">B Link: //www.cdc.gov/az/b.html </a></del></li><del> </del><li><del><a href=\"//www.cdc.gov/az/c.html\">C Link: //www.cdc.gov/az/c.html </a></del></li>
...
```

Note how the first `<li>` doesn’t get closed properly… but actually, the first deletion (after all the additions), which is from that same _first_ `<li>`, happens outside an `<li>` entirely and is followed by  the close tag for the first `<li>` in the list.

This list-of-links/open-and-close-tag-nesting issue seems to be a fairly common pattern and problem.

We need to reduce this to a clearer and simpler test case to run against all our differs.

**Comment by weatherpattern** [comment]
@Mr0grog i'm working on this ticket and create a general test case for this.
Question: where (or which dates) did you get your example snippet? When I inspect the source, I don't see any changes to the a-z menu.

**Comment by Mr0grog** [comment]
Should be the exact dates in that URL:

`https://api.monitoring.envirodatagov.org/api/v0/pages/376d7f27-fa20-448a-a5ab-0ec0e78aa1ff/changes/f96bf124-57fd-46e8-a2e4-7aeef185d4ea..db9e5f92-8d16-4038-9b88-27ce723396ed` would be

Version `f96bf124-57fd-46e8-a2e4-7aeef185d4ea` (2016-10-13 04:21:38.000Z) and
Version `db9e5f92-8d16-4038-9b88-27ce723396ed` (2016-12-16 17:51:47.000Z)

**Comment by Mr0grog** [comment]
![screen shot 2017-12-09 at 10 05 51 am](https://user-images.githubusercontent.com/74178/33798162-c47c1df4-dcc8-11e7-9ddc-ad6636a88b8f.png)

**Comment by Mr0grog** [comment]
That’s what I’m seeing, at least (any specifics are lost to the mists of my months ago memory). Looks like they just made the links always `https` instead of whatever protocol you would be viewing the page with.

**Comment by weatherpattern** [comment]
Thanks, screen shot and explanation of the api url is helpful. The <ins> and <ins> and <del> tags in your code snippet was throwing me off. I added some added some tests in the PR. Feedback welcome!

**Comment by Mr0grog** [comment]
This was done in #135 and merged in 9dc46b35dfd768be0c6985a86dcfda086a8e0cfd
